packages/discord.js/src/managers/GuildEmojiManager.js

```diff
@@ -100,6 +100,45 @@ class GuildEmojiManager extends BaseGuildEmojiManager {
     for (const emoji of data) emojis.set(emoji.id, this._add(emoji, cache));
     return emojis;
   }
+
+  /**
+   * Deletes an emoji.
+   * @param {EmojiResolvable} emoji The Emoji resolvable to delete
+   * @param {string} reason Reason for deleting the emoji
+   * @returns {?Promise<GuildEmoji>}
+   */
+  async delete(emoji, reason) {
+    const resolvedEmoji = this.resolve(emoji);
+    if (!resolvedEmoji) throw new TypeError('EMOJI_RESOLVE');
+    await this.client.api.guilds(this.guild.id).emojis(resolvedEmoji.id).delete({ reason });
+    return resolvedEmoji;
+  }
+
+  /**
+   * Edits an emoji.
+   * @param {EmojiResolvable} emoji The Emoji resolvable to edit
+   * @param {GuildEmojiEditData} data The new data for the emoji
+   * @param {string} [reason] Reason for editing this emoji
+   * @returns {Promise<GuildEmoji>}
+   */
+  async edit(emoji, data, reason) {
+    const resolvedEmoji = this.resolve(emoji);
+    if (!resolvedEmoji) throw new TypeError('EMOJI_RESOLVE');
+    const roles = data.roles?.map(r => r.id ?? r);
```

Comment:
This should use `guild.roles.resolveId`.

Reply:
Changed this too, thanks for pointing it out